packages/gatsby-plugin-gatsby-cloud/src/indicator.js

```diff
@@ -0,0 +1,342 @@
+        fill="white"
+      />
+    </svg>
+  </>
+)
+
+export default class Indicator extends React.Component {
+  state = {
+    attributes: {},
+    icon: ``,
+  }
+
+  componentDidMount() {
+    const prettyUrlRegex = /^preview-/
+    const host = window.location.hostname
+    let buildId
+
+    try {
+      setInterval(
+        async () => {
+          // currentBuild is the most recent build that is not QUEUED
+          // latestBuild is the most recent build that finished running (ONLY status ERROR or SUCCESS)
+          const { currentBuild, latestBuild } = await getBuildInfo()
+
+          if (!buildId) {
+            if (prettyUrlRegex.test(host)) {
+              buildId = latestBuild?.id
+            } else {
+              const buildIdMatch = host.match(/build-(.*?(?=\.))/)
+              buildId = buildIdMatch && buildIdMatch[1]
+            }
+          }
+
+          if (currentBuild?.buildStatus === `BUILDING`) {
+            this.setState(prevState =>
+              Object.assign({}, prevState, {
+                attributes: indicatorSetBuilding(),
+                icon: <Spinner />,
+              })
+            )
+          } else if (currentBuild?.buildStatus === `ERROR`) {
+            this.setState(prevState =>
+              Object.assign({}, prevState, {
+                attributes: indicatorSetFailed(),
+                icon: <FailedIcon />,
+              })
+            )
+          } else if (buildId === currentBuild?.id) {
+            this.setState(prevState =>
+              Object.assign({}, prevState, {
+                attributes: indicatorSetUpToDate(),
+                icon: ``,
+              })
+            )
+          } else if (
+            buildId !== latestBuild?.id &&
+            latestBuild?.buildStatus === `SUCCESS`
+          ) {
+            this.setState(prevState =>
+              Object.assign({}, prevState, {
+                attributes: indicatorSetSuccess(latestBuild?.id),
+                icon: <SuccessIcon />,
+              })
+            )
+          }
+        },
+        process.env.NODE_ENV === `test` ? 10 : POLLING_INTERVAL
+      )
+    } catch (e) {
+      console.log(e)
+    }
+  }
+
+  render() {
+    if (Object.keys(this.state.attributes).length === 0) {
+      return <>{this.props.children}</>
+    } else {
+      return (
+        <>
+          <Style />
+          <div
+            data-testid="preview-status-indicator"
+            onClick={this.state.attributes.onclick}
+            style={{
+              color: this.state.attributes.color,
+              backgroundColor: this.state.attributes.backgroundColor,
+              cursor: this.state.attributes.cursor,
+            }}
+            data-gatsby-loading-indicator="root"
+            data-gatsby-loading-indicator-visible={
+              this.state.attributes.visible
+            }
+            aria-live="assertive"
```

Comment:
We should make sure that screen readers still correctly announce it in this setup (without react portal but with wrapPageElement). If you want we can pair on this

Reply:
Thank you for all these suggestions @LekoArts! We're hoping to get the actual design for this the end of this week/early next week. I'm hesitant to spend time fixing up the styles/html because it might have to be completely redone regardless. But I'd love to pair with you once I take a stab at implement the changes to make sure that they're as good as can be :)

Reply:
@wardpeet I was not familiar with using React Portal so I used the easiest set up for me to get up and running. I'm happy to refactor it when we implement the new UI design. Will this noticeably negatively impact the performance, as in, should we wait to ship this until we're using React Portal or is it okay to ship this implementation now as is and refactor it down the line?

Reply:
Alpha-beta stage, this is totally fine. Production not so much so we'll handle this when the new design is ready. We can pair on it together.